kubert/src/log.rs

```diff
@@ -16,12 +23,129 @@ pub enum LogFormat {
     Json,
 }
 
+/// Configures the global default tracing filters.
+///
+/// A cloneable version of [`tracing_subscriber::EnvFilter`].
+#[derive(Clone, Debug)]
+#[cfg_attr(docsrs, doc(cfg(feature = "log")))]
+pub struct LogFilter(Arc<EnvFilter>);
+
 /// Indicates that an invalid log format was specified
 #[derive(Debug, Error)]
 #[error("invalid log level: {0} must be 'plain' or 'json'")]
 #[cfg_attr(docsrs, doc(cfg(feature = "log")))]
 pub struct InvalidLogFormat(String);
 
+// ==== impl LogFilter ===
+
+impl LogFilter {
+    /// Returns a new `LogFilter` from the value of the `RUST_LOG` environment
+    /// variable, ignoring any invalid filter directives.
+    ///
+    /// If the environment variable is empty or not set, or if it contains only
+    /// invalid directives, a default directive enabling the [`ERROR`] level is
+    /// added.
+    ///
+    /// [`ERROR`]: tracing::Level::ERROR
+    #[inline]
+    pub fn from_default_env() -> Self {
+        Self(EnvFilter::from_default_env().into())
+    }
```

Comment:
might be worth having an `into_inner` method or something for this using `Arc::into_inner`...unless we don't want to commit to having these filters always internally be an `EnvFilter`.